# Disallow the assignment to `exports` (`n/no-exports-assign`)

ğŸ’¼ This rule is enabled in the following [configs](https://github.com/eslint-community/eslint-plugin-n#-configs): â˜‘ï¸ `flat/recommended`, ğŸŸ¢ `flat/recommended-module`, âœ… `flat/recommended-script`, â˜‘ï¸ `recommended`, ğŸŸ¢ `recommended-module`, âœ… `recommended-script`.

<!-- end auto-generated rule header -->

To assign to `exports` variable would not work as expected.

```js
// This assigned object is not exported.
// You need to use `module.exports = { ... }`.
exports = {
    foo: 1
}
```

## ğŸ“– Rule Details

This rule is aimed at disallowing `exports = {}`, but allows `module.exports = exports = {}` to avoid conflict with [n/exports-style](https://github.com/eslint-community/eslint-plugin-n/tree/master/docs/rules/exports-style.md) rule's `allowBatchAssign` option.

ğŸ‘ Examples of **correct** code for this rule:

```js
/*eslint n/no-exports-assign: error */

module.exports.foo = 1
exports.bar = 2

module.exports = {}

// allows `exports = {}` if along with `module.exports =`
module.exports = exports = {}
exports = module.exports = {}
```

ğŸ‘ Examples of **incorrect** code for this rule:

```js
/*eslint n/no-exports-assign: error */

exports = {}
```

## ğŸ” Implementation

- [Rule source](../../lib/rules/no-exports-assign.js)
- [Test source](../../tests/lib/rules/no-exports-assign.js)
