# Requires the `data-test-id` attribute on <button> elements.

## Rule Details

Examples of **incorrect** code for this rule:

```html
<button>Download</button>
```

Examples of **correct** code for this rule:

```html
<button data-test-id="download-button">Download</button>
```

